test/charms/test_main/src/charm.py

```diff
@@ -170,6 +176,8 @@ def _on_ha_relation_broken(self, event: ops.RelationBrokenEvent):
             'relation-broken events cannot have a reference to a remote application')
         assert event.unit is None, (
             'relation broken events cannot have a reference to a remote unit')
+        assert not event.relation.active, "relation broken events always have a broken relation"
```

Comment:
Micro-nit: would be good if we can be consistent about quoting style (single quotes in this case). Also in `ops/main.py:405` and `test/test_testing.py:610`.

Should we (in a separate PR) use flake8's quoting support to automatically do this? (There are several inconsistencies in the existing codebase we'll have to fix.) https://pypi.org/project/flake8-quotes/ ... Or I could just stop noticing it. :-)

Reply:
> Micro-nit: would be good if we can be consistent about quoting style (single quotes in this case). Also in `ops/main.py:405` and `test/test_testing.py:610`.

Fixed.

> Should we (in a separate PR) use flake8's quoting support to automatically do this? (There are several inconsistencies in the existing codebase we'll have to fix.) https://pypi.org/project/flake8-quotes/ ... Or I could just stop noticing it. :-)

We should definitely do one of those, yes :joy:. I very strongly prefer to have a tool that either auto-fixes or auto-checks style rules, rather than relying on documented (or worse, undocumented) style guides. I'm also in favour of consistency, so having a linter wins over not noticing imo :smile:.

My habit is double-quotes - I think because I came to Python from predominately C/C++, but also (if I recall correctly) most of the Python developers I interacted with early on were double-quote users. Then black picking double-quotes just solidified that for me more. I'll eventually break the habit, but having tox remind me will help!

(I do see the arguments for single-quotes being preferable, particularly that `repr(string)` prefers single, so the language 'blesses' that form, and not needing shift on most English keyboard layouts. My defaulting to double is habit, not because I believe it's better/more correct).

Reply:
Item opened on adding a linting rule: https://github.com/canonical/operator/issues/1099